pyomo/core/base/constraint.py

```diff
@@ -386,8 +390,10 @@ def upper(self):
     @property
     def lb(self):
         """Access the value of the lower bound of a constraint expression."""
-        bound = value(self._lb())
-        if bound is not None and not math.isfinite(bound):
+        bound = self._lb()
+        if bound.__class__ not in native_types:
+            bound = value(bound)
+        if bound in _nonfinite_values or bound != bound:
```

Comment:
When would `bound != bound` ever trigger?

Reply:
If bound is `float('nan')`, then `(bound != bound) is True`

Reply:
(this and the other three instances of this idiom are covered by tests in `pyomo/core/tests/unit/test_con.py`)